app/models/user.rb

```diff
@@ -0,0 +1,27 @@
+class User < ActiveRecord::Base
+  has_many :nono_users
+  has_many :nonos, through: :nono_users
+  has_many :rsvps
+  has_many :parties, through: :rsvps
+
+  has_secure_password
+
+  validates :email, uniqueness: true, presence: true
+  validates_format_of :email, :with => /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i, :on => :create
+  validates :name, presence: true
+  validates :password, confirmation: true, length: { minimum: 6 }
+  validates :password_confirmation, presence: true
+
+  def on_guest_list?(party)
+    guest_list = party.users
+    guest_list.include?(self)
+  end
+
+  def upcoming_parties
+    self.parties.select{|party| party.upcoming?}
```

Comment:
This would be better as a database query - otherwise you are loading all parties ever and filtering out the past ones. That's probably not going to be a great idea when your site has been up 10 years.